include/diaspora.php

```diff
@@ -701,6 +701,33 @@ public static function handle_from_contact($contact_id, $gcontact_id = 0) {
 	}
 
 	/**
+	 * @brief get a url (scheme://domain.tld/u/user) from a given Diaspora
+	 * fcontact id or guid
+	 *
+	 * @param mixed $contact_id Either the numeric id or the string guid
+	 *
+	 * @return string the url
+	 */
+	public static function url_from_fcontact($contact_id) {
+		$handle = False;
+
+		logger("fcontact id is ".$contact_id, LOGGER_DEBUG);
+
+		if (is_numeric($contact_id)) {
```

Comment:
This is only possible because PHP assigns variable/parameter types lacily. If one day this changes, this code will no longer work. Better is to have 2 methods around, e.g. `url_from_fcontact_by_id()` and `url_from_fcontact_by_guid()`. You may duplicate a bit code but it is much more clear code.